src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -3938,6 +3940,9 @@ def _check_running(self):
             else:
                 self._prof.prof('exec', msg='execution complete', uid=cu['_id'])
 
+                # make sure proc is collected
+                cu['proc'].wait()
```

Comment:
yes, this is actually the one I stumbled over.  It is probably redundant to the popen fix now though,  granted.  This one does not fix the popen problem we had in its own, but made sure that the pid is collected.  

Since it is redundant, I don't mind removing it, but I would also not mind leaving it for its explicity.